csharp/src/Microsoft.ML.OnnxRuntime/CheckpointState.shared.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.OnnxRuntime
+{
+    /// <summary>
+    ///  Sets various runtime options. 
+    /// </summary>
+    public class CheckpointState : SafeHandle
+    {
+        internal IntPtr Handle
+        {
+            get
+            {
+                return handle;
+            }
+        }
+
+        /// <summary>
+        /// Default __ctor. Creates default RuntimeOptions
+        /// </summary>
+        public CheckpointState()
+            : base(IntPtr.Zero, true)
+        {
+            var envHandle = OrtEnv.Handle; // just so it is initialized
+        }
+
+        /// <summary>
+        /// Overrides SafeHandle.IsInvalid
+        /// </summary>
+        /// <value>returns true if handle is equal to Zero</value>
+        public override bool IsInvalid { get { return handle == IntPtr.Zero; } }
+
+        public void LoadCheckpoint(string checkpointPath)
+        {
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtLoadCheckpoint(NativeMethods.GetPlatformSerializedString(checkpointPath), out handle));
```

Comment:
>GetPlatformSerializedString [](http://example.com/codeflow?start=88&length=27)

I believe, this is wrong. C API is stable regardless of the platform as specified in the NativeMethods documentation for the API.
As such it always prerequires UTF-8 C null terminated string. The string must be converted to UTF-8.